src/azure-cli-core/azure/cli/core/telemetry.py

```diff
@@ -217,85 +214,59 @@ def _wrapper(*args, **kwargs):
 # public api
 
 
-@decorators.suppress_all_exceptions(raise_in_diagnostics=True)
+@decorators.suppress_all_exceptions()
 def start(mode=None):
     if mode:
         _session.mode = mode
     _session.start_time = datetime.datetime.utcnow()
 
 
 @_user_agrees_to_telemetry
-@decorators.suppress_all_exceptions(raise_in_diagnostics=True)
+@decorators.suppress_all_exceptions()
 def flush():
+    from azure.cli.core._environment import get_config_dir
+    from azure.cli.telemetry import save
+
     # flush out current information
     _session.end_time = datetime.datetime.utcnow()
-
-    payload = _session.generate_payload()
-    if payload:
-        _upload_telemetry_with_user_agreement(payload)
+    save(get_config_dir(), _session.generate_payload())
 
     # reset session fields, retaining correlation id and application
     _session.__init__(correlation_id=_session.correlation_id, application=_session.application)
 
 
 @_user_agrees_to_telemetry
-@decorators.suppress_all_exceptions(raise_in_diagnostics=True)
+@decorators.suppress_all_exceptions()
 def conclude():
-    _session.end_time = datetime.datetime.utcnow()
+    from azure.cli.core._environment import get_config_dir
+    from azure.cli.telemetry import save
 
-    payload = _session.generate_payload()
-    if payload:
-        kwargs = dict()
-        if os.name == 'nt':
-            # Windows process creation flag to not reuse the parent console.
-            # Without this, the background service is associated with the
-            # starting process's console, and will block that console from
-            # exiting until the background service self-terminates.
-            # Elsewhere, fork just does the right thing.
-            kwargs['creationflags'] = 0x00000010  # CREATE_NEW_CONSOLE
-
-            startupinfo = subprocess.STARTUPINFO()
-            startupinfo.dwFlags |= subprocess.STARTF_USESHOWWINDOW
-            startupinfo.wShowWindow = subprocess.SW_HIDE
-            kwargs['startupinfo'] = startupinfo
-        else:
-            if sys.version_info >= (3, 3):
-                kwargs['stdin'] = subprocess.DEVNULL
-                if not telemetry_core.in_diagnostic_mode:
-                    kwargs['stdout'] = subprocess.DEVNULL
-                    kwargs['stderr'] = subprocess.STDOUT
-        _upload_telemetry_with_user_agreement(payload, **kwargs)
-
-
-# This includes a final user-agreement-check; ALL methods sending telemetry MUST call this.
-@_user_agrees_to_telemetry
-@decorators.suppress_all_exceptions(raise_in_diagnostics=True)
-def _upload_telemetry_with_user_agreement(payload, **kwargs):
```

Comment:
Removing this removes the extra check for user agreement, just so you're aware.

Reply:
yes